src/main/services/request-auth.ts

```diff
@@ -32,38 +32,45 @@ export class AutomationCloudAuthService extends RequestAuthService {
         protected acContext: AutomationCloudContext,
     ) {
         super();
-        const baseUrl = this.env.API_AUTH_URL; // for forwardRequestHeader
+        const baseUrl = this.env.AC_AUTH_MIDDLEWARE_URL;
         this.clientRequest = new Request({
             baseUrl,
             retryAttempts: 3,
         });
     }
 
     async check(ctx: Koa.Context) {
+        const token = await this.getTokenFromHeader(ctx.req.headers as any);
+        if (token) {
+            const organisationHeader = ctx.req.headers['x-ubio-organisation-id'] as string;
+            await this.verifyAuth(token, organisationHeader);
+        }
+    }
+
+    protected async getTokenFromHeader(headers: { [name: string]: string | undefined }) {
         const authHeaderName = this.env.AC_AUTH_HEADER_NAME;
-        const newAuthHeader = ctx.req.headers[authHeaderName] as string;
+        const newAuthHeader = headers[authHeaderName];
         if (newAuthHeader) {
-            const organisationHeader = ctx.req.headers['x-ubio-organisation-id'] as string;
-            await this.handleNewAuth(newAuthHeader, organisationHeader);
-            return;
+            const [prefix, token] = newAuthHeader.split(' ');
+            if (prefix !== 'Bearer' || !token) {
+                throw new Exception({
+                    name: 'AuthenticationError',
+                    message: `Incorrect authorization header (prefix=${prefix}, tokenExists=${!!token})`,
+                    status: 401
+                });
+            }
+            return token;
         }
-        // Fallback to legacy header forwarding
-        const legacyAuthHeader = ctx.req.headers['authorization'];
-        if (legacyAuthHeader) {
-            await this.handleLegacyAuth(legacyAuthHeader);
+
+        const legacyHeader = headers['authorization'];
+        if (legacyHeader) {
+            return this.handleLegacyAuth(legacyHeader);
         }
     }
 
-    protected async handleNewAuth(newAuthHeader: string, organisationHeader: string) {
-        const [prefix, token] = newAuthHeader.split(' ');
-        if (prefix !== 'Bearer' || !token) {
-            throw new Exception({
-                name: 'AuthenticationError',
-                message: `Incorrect authorization header (prefix=${prefix}, tokenExists=${!!token})`,
-                status: 401
-            });
-        }
+    protected async verifyAuth(token: string, organisationHeader: string) {
         try {
+            // TODO: add cache
```

Comment:
No need, jwks should be caching the keys, other than that I don't think we need anything else cached